include/seqan3/core/algorithm/configuration.hpp

```diff
@@ -19,6 +19,7 @@
 #include <seqan3/core/algorithm/concept.hpp>
 #include <seqan3/core/algorithm/configuration_utility.hpp>
 #include <seqan3/core/algorithm/pipeable_config_element.hpp>
+#include <seqan3/core/type_list/all.hpp>
```

Comment:
This will already  include the `type_traits/basic.hpp` and `type_traits/template_inspection.hpp` headers, so you could remove these two.

Also applies to other files.

Reply:
Hmmm but I thought they should stay because it should be self contained? Like in seqan3/alphabet/concept.hpp there is an include for seqan3/core/concept/core_language.hpp but we include both usually right?

Reply:
I'm in favour of minimising the includes throughout the library and only be more verbose in the snippets since this is public documentation.
Until we have clang support and consequently also iwyu support I'd at least not deliberately bloat the include list with redundant includes...

Reply:
Okay!